src/draw_functions.py

```diff
@@ -81,8 +81,9 @@ def levels(sample, cr, width, height, colors, offset, radius, fill, thickness):
     step = width / ls
     offset_px = step * offset / 100
     if not fill:
+        thickness = thickness * (100 - offset) / 100
```

Comment:
I disagree with this change. I already made elements change depending on thickness (see lines 90-99 for example, when fill is off y coordinate and height are calculated taking thickness into account).

Reply:
Thickness is set in pixels, so it's unexpected if it will change depending on something else.
But I like how in the last commit you changed in bars mode so when fill is on it just change thickness to 0 and one function is used no matter what in what state fill is, shorter and cleaner code.

Reply:
The problem I was trying to fix was that if you put thickness to full, and then increased offset, you would see the blank space inside the element shrink to nothing, until it at some point grows again slightly

Reply:
[Kooha-2023-02-04-17-10-45.webm](https://user-images.githubusercontent.com/119804311/216777732-ec350851-4c77-4904-8a97-c683113e53f3.webm)

Like this.

Reply:
> The problem I was trying to fix was that if you put thickness to full, and then increased offset, you would see the blank space inside the element shrink to nothing, until it at some point grows again slightly

I see another symptoms, elements start to grow and overlap eachother if both thickness and offset are big enough.
We need to fix it in some different way, without modifying thickness when the problem is not happening.

Reply:
Even without offset, when thickness is too big elements overlap. Tested with particles mode.

Reply:
> Even without offset, when thickness is too big elements overlap. Tested with particles mode.

I've tried to do the same with particles mode, which fixes it. Is it really so bad to use offset?

Reply:
I found a fix for particles mode, I just limited thickness to not be more than half of minimum size of an element. I assume the same should work for other modes, going to implement it all and commit then.

Reply:
> Is it really so bad to use offset?

Yes. Thickness is in pixels. If a user set it to be 20, it should be 20, unless it causes problems.

Reply:
Can you try again with my latest commit? It fixed it for me, but thickness calculation includes step & offset.

Reply:
> > Is it really so bad to use offset?
> 
> Yes. Thickness is in pixels. If a user set it to be 20, it should be 20, unless it causes problems.

Well, we could also change the description of thickness... 😬

Reply:
> Well, we could also change the description of thickness... grimacing

Or we can make it work properly :) I found a way to fix particles mode when roundness is 0, but when I increase roundness it doesn't work for some reason. I'm goind to find why and fix it